### PR TITLE
Fixed memory leak in png_handle_eXIf (#220).

### PR DESCRIPTION
**Pull request by dlemstra**
This patch fixes the memory leak reported in #220.

### CODE REVIEW

**Comment by dlemstra** [comment]
Just did a rebase to fix this PR. Would love to finally see this get merged.

File: pngrutil.c [hunk 1]
```diff
@@ -2087,10 +2087,8 @@ png_handle_eXIf(png_structrp png_ptr, png_inforp info_ptr, png_uint_32 length)
       }
    }
 
-   if (png_crc_finish(png_ptr, 0) != 0)
-      return;
-
-   png_set_eXIf_1(png_ptr, info_ptr, length, info_ptr->eXIf_buf);
+   if (png_crc_finish(png_ptr, 0) == 0)
+     png_set_eXIf_1(png_ptr, info_ptr, length, info_ptr->eXIf_buf);
```
Comment (alalek) [inline pngrutil.c]: please use 3 spaces indentation (like other code in this file)

Comment (dlemstra) [reply pngrutil.c]: Just did a rebase and a new commit to fix this.

**Comment by tangyaofang** [comment]
The perfect solution, memory leaks should occur with if (png_crc_finish(png_ptr, 0) != 0).

**Comment by dlemstra** [comment]
Do you want me to do anything else @ctruta or can we merge this?

**Comment by ctruta** [comment]
Hi, Dirk. As I was going through the pull requests in a completely non-scientific order, I missed the fact that your commit was submitted way before Skal's. I should have applied your fix. Many apologies.

As I am preparing (finally!) the libpng-1.6.38 release, I will give credit to both you and Skal.

**Pull request closed without merging**